sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -614,7 +614,11 @@ object ViewHelper extends SQLConfHelper with Logging {
     }.getOrElse(false)
     if (replace && uncache) {
       logDebug(s"Try to uncache ${name.quotedString} before replacing.")
-      checkCyclicViewReference(analyzedPlan, Seq(name), name)
+      if (!conf.storeAnalyzedPlanForView && originalText.nonEmpty) {
```

Comment:
maybe it's better to have a `val storeAnalyzedPlanForView = conf.storeAnalyzedPlanForView || originalText.nonEmpty` at the beginning, then we can just write `if (!storeAnalyzedPlanForView)` here and below.